syntax_and_semantics/capturing_blocks.md

```diff
@@ -2,20 +2,34 @@
 
 A block can be captured and turned into a `Proc`, which represents a block of code with an associated context: the closured data.
 
-To capture a block you must specify it as a method's block argument, give it a name and specify the input and output types. For example:
+A block is captured if these conditions are met:
+- The block is specified as a method's block argument (e.g. `def method(&block)`)
+- The input(s) and output are specified (e.g. `def method(&block : Int32 -> Int32)`)
+- The block is used in the body of the method (returning the block is sufficient)
+
+The input(s) and/or output types can be omitted - meaning there is no input(s) or no output, respectively. Input types must be specific (e.g. `Int32`, `String`) while the output can be a specific type or generic (using `_`).
+
+## Simple example
 
 ```crystal
 def int_to_int(&block : Int32 -> Int32)
   block
 end
 
 proc = int_to_int { |x| x + 1 }
-proc.call(1) #=> 2
+proc.call(1) # => 2
 ```
 
 The above code captures the block of code passed to `int_to_int` in the `block` variable, and returns it from the method. The type of `proc` is [Proc(Int32, Int32)](http://crystal-lang.org/api/Proc.html), a function that accepts a single `Int32` argument and returns an `Int32`.
 
-In this way a block can be saved as a callback:
+The simple example covers all requirements for capturing a block:
+- `def int_to_int(&block : Int32 -> Int32)` - specify it as a method's block argument
+- `def int_to_int(&block : Int32 -> Int32)` - specify the input(s) and output
```

Comment:
As I said, this isn't needed. If you do:

```crystal
def foo(&block)
  block
end
```

that defines a method that captures a block that has no inputs and returns `nil`.

Reply:
"requirements" is probably the wrong word, but I want to show that all those things are happening either explicitly or implicitly when a block is captured.

In your example, the "no inputs" and `nil` return are implied. So I need to find a better way to express that in the docs.